vendor_pricelist_category/models/product_supplierinfo_category.py

```diff
@@ -70,13 +71,26 @@ class ProductSupplierinfoCategory(models.Model):
         recursive=True,
     )
 
-    _sql_constraints = [
-        (
-            "code_company_unique",
-            "unique(code, company_id)",
-            _("The code is already in use."),
-        )
-    ]
+    @api.constrains("code", "company_id", "parent_id")
+    def _check_code(self):
+        for category in self.filtered("code"):
+            categories = (
+                self.env["product.supplierinfo.category"]
+                .search(
+                    [
+                        ("parent_id", "=", category.parent_id.id),
```

Comment:
Are you only searching at the same level? You don't have to search the entire category hierarchy?

Reply:
only at the same level